### PR TITLE
Better errors for unfilled inputs

### PR DESCRIPTION
**Pull request by Bjvanminnen**
Here are a couple examples of error messages we'll see now. As I look at these, I also wonder if we should change the icon for calc/eval.

image

image

image

### CODE REVIEW

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -1599,8 +1608,46 @@ StudioApp.prototype.hasUnfilledFunctionalBlock = function () {
...
+  if (unfilled.type === topLevelType) {
+    return msg.emptyTopLevelBlock({topLevelBlockName: unfilled.getTitleValue()});
+  }
+
+  if (topParent.type !== 'functional_definition') {
+    return msg.emptyFunctionalBlock();
+  }
+
+  var procedureInfo = topParent.getProcedureInfo();
+  var isVariable = procedureInfo.parameterNames.length === 0;
```
Comment (Bjvanminnen) [inline apps/src/StudioApp.js]: Is there a better way to check for this?

**Comment by tanyaparker** [comment]
Icon as in the dog or pencil dude? We plan on removing these next sprint.

Can Big Game also be "Try again" instead of "Keep playing?"

**Comment by Bjvanminnen** [comment]
Yes, referring to pencil dude. Removing makes sense.

Will do on replacing Try again with Keep playing. Do you want this just for Big Game, or for all studio levels?

**Comment by Bjvanminnen** [comment]
Looks like the current heuristic in studio is Keep Playing if freeplay, try again otherwise. I think all Big Game levels are freeplay, so we might already be set.

**Comment by Bjvanminnen** [comment]
image

File: apps/src/StudioApp.js (hunk 1 above)
Comment (bcjordan) [inline apps/src/StudioApp.js]: There's a `topParent.isVariable()`

**Comment by tanyaparker** [comment]
All error messages for all calc/eval/all studio should be "Try again"

Even though Big Game is freeplay, we do want them to achieve certain goals. So "Try again" makes more sense to reinforce that they need to fix something. Can we override it for Big Game?

File: apps/src/StudioApp.js [hunk 2]
```diff
@@ -1590,7 +1590,16 @@ StudioApp.prototype.hasQuestionMarksInNumberField = function () {
  * @returns true if any non-example block in the workspace has an unfilled input
  */
 StudioApp.prototype.hasUnfilledFunctionalBlock = function () {
-  return Blockly.mainBlockSpace.getAllBlocks().some(function (block) {
+  return !!this.getUnfilledFunctionalBlock();
```
Comment (bcjordan) [inline apps/src/StudioApp.js]: Always a fan of the single point of truthin' :+1:

**Comment by Bjvanminnen** [comment]
Yeah, we can override it. I got confused and was thinking we wanted keep playing vs. the other way around. I'll fix this.

**Comment by bcjordan** [comment]
LGTM

**Pull request merged**